contracts/governance/Vesting/VestingRegistryStorage.sol

```diff
@@ -61,4 +61,17 @@ contract VestingRegistryStorage is Initializable, AdminRole {
     ///@notice A record of all vesting addresses
     ///@dev isVesting[address] returns if the address is a vesting address
     mapping(address => bool) public isVesting;
+
+    /// @notice Store vesting creation type & vesting type information
+    /// @dev it is packed into 1 single storage slot for cheaper gas usage
+    struct VestingCreationAndTypeDetail {
+        bool isSet;
+        uint120 vestingType;
+        uint128 vestingCreationType;
+    }
+
+    ///@notice A record of all vesting addresses with the detail
+    ///@dev vestingDetail[vestingAddress] returns Vesting struct data
+    ///@dev can be used to easily check the vesting type / creation type based on the vesting address itself
+    mapping(address => VestingCreationAndTypeDetail) public vestingCreationAndTypes;
```

Comment:
```suggestion 
    mapping(address => VestingCreationAndTypeDetails)
```

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/7f1c269ee40b1d2f3f864b07c574360839a9d270)